src/Extension/CoreExtension.php

```diff
@@ -911,31 +911,91 @@ function twig_in_filter($value, $compare)
         $value = (string) $value;
     }
 
-    if (\is_array($compare)) {
-        return \in_array($value, $compare, \is_object($value) || \is_resource($value));
-    } elseif (\is_string($compare) && (\is_string($value) || \is_int($value) || \is_float($value))) {
-        return '' === $value || false !== strpos($compare, (string) $value);
-    } elseif ($compare instanceof \Traversable) {
-        if (\is_object($value) || \is_resource($value)) {
-            foreach ($compare as $item) {
-                if ($item === $value) {
-                    return true;
-                }
-            }
-        } else {
-            foreach ($compare as $item) {
-                if ($item == $value) {
-                    return true;
-                }
-            }
+    if (\is_string($compare)) {
+        if (\is_string($value) || \is_int($value) || \is_float($value)) {
+            return '' === $value || false !== strpos($compare, (string) $value);
         }
 
         return false;
     }
 
+    if (\is_object($value) || \is_resource($value)) {
```

Comment:
this is missing the `is_array($compare)` check. Calling `in_array` on a Traversable won't work fine AFAICT.

Reply:
@fabpot what about this one ?

Reply:
I'm moved the `is_iterable` check above this condition, does that work for you?

Reply:
nope. We still don't want to pass iterators there.

Reply:
and tests are not catching this, because the case of `value in iterator` is not covered by the tests.

Reply:
fixed, test added